Testing/PerformanceTests/analysis.py

```diff
@@ -115,147 +129,85 @@ def get_results_matching(results, field, value):
-        space_available +=spacing
-        if space_available >= s_width:
-            space_available = 0
-            tick_index.append(i)
-            tick_strings.append(s)
-
-    xticks( tick_index, tick_strings )
-
-
-
-#============================================================================================
-def plot_success_count(type='system', last_num=-1, x_field='revision'):
-    """ Plot the count of successful/failed tests vs revision number
-
-    Parameters
-    ----------
-        type :: 'system', or 'performance'
-    """
-    results = get_results('', type, where_clause='', orderby_clause=get_orderby_clause(last_num))
-    revisions = get_unique_fields(results, x_field)
-
-    # Go through each revision
-    success = []
-    fail = []
-    for revision in revisions:
-        these = get_results_matching(results, x_field, revision)
-        succeeded = 0
-        failed = 0
-        for res in these:
-            if res["success"]:
-                succeeded += 1
-            else:
-                failed += 1
-        # Keep the list of them
-        success.append(succeeded)
-        fail.append(failed)
-
-    figure()
-    revisions = np.array(revisions)
-    fail = np.array(fail)
-    success = np.array(success)
-
-    index = np.arange(len(revisions))
-#    p1 = bar(index, fail, color='r')
-#    p2 = bar(index, success, color='g', bottom=fail)
-#    legend( (p1[0], p2[0]), ('Failure', 'Success') )
-
-    p1 = fill_between(index, fail, 0, color='r')
-    p2 = fill_between(index, success+fail, fail, color='g')
-    #legend( (p1, p2), ('Failure', 'Success') )
-
-    smart_ticks( index, revisions)
-
-
-    ylabel('Success/Fail')
-    xlabel(x_field)
-    revsare = "all revs"
-    if last_num > 0: revsare = "last %d revs" % last_num
-    title("Success/Fail History of %s tests (%s)" % (type, revsare))
-
-
-#============================================================================================
-def plot_runtime(*args, **kwargs):
-    """ Call get_runtime_data()
-
-    Parameters
-    ----------
-        - See get_runtime_data() for the full list
-    """
-    (x,y) = get_runtime_data(*args, **kwargs)
-
-    figure()
-    index = np.arange(len(x))
-    plot(index,y,'-b.')
-    smart_ticks( index, x)
-    ylabel('Runtime/iteration (sec)')
-    xlabel(kwargs['x_field'])
-
-    last_num =kwargs.get('last_num',-1)
+# ============================================================================================
+def plot_runtime(annotate, saveImage, path, **kwargs):
+    name = kwargs['name']
+    (xData, yData) = get_runtime_data(**kwargs)
+    trace1 = go.Scatter(x=xData, y=yData,
+                        mode='lines+markers',
+                        marker=dict(
+                            size='5',
+                            color="blue"
+                        )
+                        )
+
+    annotations = []
+
+    last_num = kwargs.get('last_num', -1)
```

Comment:
`None` is generally a better "invalid or absent value" placeholder in Python, since we're not restricted by type.

The check a bit further down would then become:
```python
if last_num is not None:
    commitids = commitids[-last_num:]
```
This would then also support `last_num == 0` for getting all results in spite of providing a last_num, and negative `last_num` values to mean "skip the first (-)X items" instead of "give me the last X items". 

Another alternative with the same effect would be to forgo having a `last_num` local variable and just do:
```python
if 'last_num' in kwargs:
    commitids = commitids[-kwargs['last_num']:]  # or kwargs.get('last_num')
```
Although I find this a bit less nice to read.